README.md

```diff
@@ -1,4 +1,140 @@
+Microcluster is a Go library that provides an easy to use framework for creating and managing highly available [dqlite](https://github.com/canonical/dqlite) clusters. It provides an extensible API and database which can be used to directly connect to your underlying services.
+
+Extend Microcluster with your own API endpoints, schema and API extensions, lifecycle actions, and additional network listeners.
+
+## Import Microcluster
+
+Get the latest LTS release of Microcluster:
+
+```
+$ go get github.com/canonical/microcluster/v2@latest
+```
+
+## Usage
+For a step-by-step practical example of creating a Microcluster project, check out the [example package](example). Examine the code or build and test it out with the included Makefile.
+
+All of Microcluster's state is stored in the `StateDir` directory. Define a new Microcluster app:
+
+```go
+m, err := microcluster.App(microcluster.Args{StateDir: "/path/to/state"})
+if err != nil {
+    // ...
+}
+```
+
+Extensions to Microcluster must be defined in `DaemonArgs`:
+```go
+
+dargs := microcluster.DaemonArgs{
+    Version: "0.0.1",
+
+    ExtensionsSchema: []schema.Update{
+        // Sequential list of modifications to the database.
+    },
+
+    APIExtensions: []string{
+        // Sequential list of API capabilities.
+    },
+
+    Servers: map[string]rest.Server{
+        "my-server": {
+            CoreAPI: true,
+            ServeUnix: true,
+            Resources: []rest.Resources{
+                {
+                    PathPrefix: "1.0",
+                    Endpoints: []rest.Endpoint{
+                        // Additional API endpoints to offer on "/1.0".
+                    },
+                },
+            }
+        }
+    }
+}
+```
+
+Start the daemon with your configuration:
+```go
+err := m.Start(ctx, dargs)
+if err != nil {
+    // ...
+}
+```
+
+### Define lifecycle actions (Hooks):
+```go
+dargs.Hooks = &state.Hooks{
+    OnStart: func(ctx context.Context, s state.State) error {
+        // Execute code on each startup.
+    }
+}
+```
+
+### Query the dqlite database directly:
+```go
+
+_, batch, err := m.Sql(ctx, "SELECT name, address FROM core_cluster_members WHERE role='voter'")
+if err != nil {
+    // ...
+}
+
+for i, result := range batch.Results {
+    fmt.Printf("Query %d:\n", i)
+
+    fmt.Println("Type:", result.Type)
+    fmt.Println("Columns:", result.Columns)
+    fmt.Println("RowsAffected:", result.RowsAffected)
+    fmt.Println("Rows:", result.Rows)
+}
+
+```
+
+### Create your own API endpoints:
+```go
+endpoint := rest.Endpoint{
+    Path: "mypath" // API is served over /mypath
+
+    Post: rest.EndpointAction{Handler: myHandler} // POST action for the endpoint.
+
+func myHandler(s state.State, r *http.Request) response.Response {
+    msg := fmt.Sprintf("This is a response from %q at %q", s.Name(), s.Address())
+
+    return response.SyncResponse(true, msg)
+}
+
+// Include your endpoints in DaemonArgs like so to serve over /1.0 over the default network.
```

Comment:
```suggestion
// Include your endpoints in DaemonArgs to serve over /1.0 over the default network.
```
I'm not a huge fan of "default network"; is there a name we can use to tie `CoreAPI` and `StartAPI`'s `newConfig.Location.Address` parameter? It took me a while to figure out what CoreAPI meant; this is a spot that could use some additional clarity IMO

Reply:
In MicroCloud we refer to this as the "cluster management network"

Reply:
Yes, it's the default listener for the `/core/*` endpoints. I think I'd prefer at least "listener" rather than "network", since it's not really a "network" per se, just a handle to an addr+port listening for HTTP(S) traffic on some network interface.